src/swaps/fastbtc/FastBTCWithdrawSwapProvider.ts

```diff
@@ -0,0 +1,394 @@
+      // TODO: Why is from amount being set in every quote call? It should be in one place only (getQuotes action)
+      fromAmount: fromAmountInUnit.toFixed(),
+      toAmount: toAmountInUnit.toFixed(),
+    };
+  }
+
+  async buildSwapTx({ network, walletId, quote }: { network: Network; walletId: WalletId; quote: SwapQuote }) {
+    const client = this.getClient(network, walletId, quote.from, quote.fromAccountId);
+    const fastBtcBridge = this.getFastBtcBridge(client.chain.getProvider());
+    // Assuming valid bitcoin address
+    const fromAddressRaw = await this.getSwapAddress(network, walletId, quote.from, quote.fromAccountId);
+    const fromAddress = chains[cryptoassets.RBTC.chain].formatAddress(fromAddressRaw);
+    const toAddress = await this.getSwapAddress(network, walletId, quote.to, quote.toAccountId);
+
+    const data = await fastBtcBridge.interface.encodeFunctionData('transferToBtc', [toAddress]);
+    const swapTx = {
+      from: fromAddress,
+      to: fastBtcBridge.address,
+      value: ethers.BigNumber.from(quote.fromAmount),
+      data,
+    };
+
+    return swapTx;
+  }
+
+  async sendSwap({ network, walletId, quote }: SwapRequest) {
+    const client = this.getClient(network, walletId, quote.from, quote.fromAccountId);
+    const fastBtcBridge = this.getFastBtcBridge(client.chain.getProvider());
+    // Assuming valid bitcoin address
+    const toAddress = await this.getSwapAddress(network, walletId, quote.to, quote.toAccountId);
+
+    await this.sendLedgerNotification(quote.fromAccountId, 'Signing required to complete the swap.');
+    const data = await fastBtcBridge.interface.encodeFunctionData('transferToBtc', [toAddress]);
+    const swapTx = await client.wallet.sendTransaction({
+      to: fastBtcBridge.address,
+      value: new BN(quote.fromAmount),
+      data,
+    });
+
+    return {
+      status: 'WAITING_FOR_SEND_CONFIRMATIONS',
+      swapTx,
+      swapTxHash: swapTx.hash,
+    };
+  }
+
+  async newSwap({ network, walletId, quote }: SwapRequest) {
+    const updates = await this.sendSwap({ network, walletId, quote });
+
+    return {
+      id: uuidv4(),
+      fee: quote.fee,
+      slippage: 50,
+      ...updates,
+    };
+  }
+
+  async estimateFees({ network, walletId, quote, feePrices }: EstimateFeeRequest) {
+    const client = this.getClient(network, walletId, quote.from, quote.fromAccountId);
+    const swapTx = await this.buildSwapTx({ network, walletId, quote });
+    const gasLimit = await client.chain.getProvider().estimateGas(swapTx);
+    const fees: EstimateFeeResponse = {};
+    for (const feePrice of feePrices) {
+      const gasPrice = new BN(feePrice).times(1e9); // ETH fee price is in gwei
+      const fee = new BN(gasLimit.toString()).times(1.1).times(gasPrice);
+      fees[feePrice] = unitToCurrency(cryptoassets.RBTC, fee);
+    }
+
+    return fees;
+  }
+
+  /**
+   * TODO: this is called very often from clients, so making a network call each time will be very intensive
```

Comment:
[Pebble] We need to create a ticket for it for better tracking

Reply:
We should but practically I know we will never pick it up and it will just be more headache for the PM